src/intentManager.js

```diff
@@ -51,167 +54,90 @@ define([
+            newTag.androidIntentExtra = parseInnerTags($tag, 'extra');
+            newTag.androidIntentResponse = parseInnerTags($tag, 'response');
 
-        widget.definition = mug.p.getDefinition('androidIntentAppId');
-        widget.currentValue = (mug.intentTag) ? mug.intentTag.getAttr('path') : "";
-        
-        var input = $("<input />")
-            .attr("name", widget.id)
-            .attr("type", "text")
-            .attr('placeholder', 'Insert Android Application ID');
+            _.chain(tagXML.attributes)
+             .filter(function(attr) {
+                 return !_.contains(['id', 'class', 'xmlns:odk'], attr.nodeName);
+             })
+             .each(function(attr) {
+                 newTag.unknownAttrs[attr.nodeName] = attr.nodeValue;
+             });
 
-        widget.getControl = function () {
-            if (widget.isDisabled()) {
-                input.prop('disabled', true);
-            }
-            return input;
-        };
+            intentTags[tagId] = newTag;
+        });
 
-        widget.setValue = function (value) {
-            input.val(value);
-        };
+        return intentTags;
+    }
 
-        widget.getValue = function() {
-            return input.val();
-        };
+    function syncMugWithIntent (tags, mug) {
+        // called when initializing a mug from a parsed form
+        if (mug.__className === "AndroidIntent") {
+            var nodeID = mug.p.nodeID,
+                tag = tags.hasOwnProperty(nodeID) ? tags[nodeID] : makeODKXIntentTag(nodeID);
 
-        widget.updateValue = function () {
-            widget.mug.intentTag.setAttr('path', widget.getValue());
-        };
+            _.each(INTENT_SPECIFIC_SPECS, function (key) {
+                mug.p[key] = tag[key];
+            });
 
-        input.bind("change keyup", widget.updateValue);
-        return widget;
+            delete tags[tag.nodeID];
+        }
     }
 
-    function androidIntentExtra(mug, options) {
-        options.id = "intent-extra";
-        var widget = widgets.baseKeyValue(mug, options);
-        widget.currentValue = (mug.intentTag) ? mug.intentTag.getAttr('extra') : {};
-        widget.definition = mug.p.getDefinition('androidIntentExtra');
+    function writeIntentXML (unmappedIntentTags, xmlWriter, tree) {
+        // make sure any leftover intent tags are still kept
+        _.each(unmappedIntentTags, function (tag) {
+            writeXML(xmlWriter, tag);
+        });
 
-        widget.save = function () {
-            if (widget.mug.intentTag) {
-                widget.mug.intentTag.setAttr('extra', widget.getValidValues());
+        function getIntentMugs(node) {
+            var mug = node.getValue();
+            if (!mug || node.isRootNode) {
+                return null;
             }
-        };
+            if (mug.options.dataType === 'intent') {
+                writeXML(xmlWriter, mug.p);
+            } else {
+                return null;
+            }
```

Comment:
Unnecessary else clause, and the first conditional above could be merged with the next. Could also move this into the treeMap call:
```js
tree.treeMap(function (node) {
    if (mug && mug.options.dataType === 'intent') {
        writeXML(xmlWriter, mug.p);
    }
});
```